### DISCUSSION TITLE
How to see the current source being used in none-ls for my status line

### DISCUSSION THREAD

**Discussion by ghost** [nvimtools/none-ls.nvim#79] [Q&A]
Currently I am trying to use this command to get the current none-ls source for my status line but it is always blank. 
`local null_ls_source = "%{StringFromList(map(get(g:, 'null_ls_sources', []), 'v:val.name'))}"`

I am using this to get the current lsp server and for none-ls (null-ls) it just says 'null-ls' so it would be helpful to know if prettier for example active.
```
local get_lsp_client = function(msg)
    msg = msg or 'No Active Lsp'
    local buf_ft = vim.api.nvim_buf_get_option(0, 'filetype')
    local clients = vim.lsp.get_active_clients()
    if next(clients) == nil then
        return msg
    end
    for _, client in ipairs(clients) do
        local filetypes = client.config.filetypes
        if filetypes and vim.fn.index(filetypes, buf_ft) ~= -1 then
            return client.name
        end
    end
    return msg
end
```

What am I doing wrong?

**Comment by Zeioth** [comment]
If you use [heirline](https://github.com/rebelot/heirline.nvim), you can use the [lsp component](https://github.com/Zeioth/heirline-components.nvim/wiki/lsp%E2%80%90component) from [heirline-components](https://github.com/Zeioth/heirline-components.nvim) to achieve this.

Or if you don't mind having it float, you can use [noice.nvim](https://github.com/folke/noice.nvim), which is quite easy to setup.
